crypto/src/test/java/org/springframework/security/crypto/argon2/Argon2EncodingUtilsTests.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.crypto.argon2;
+
+import org.bouncycastle.crypto.params.Argon2Parameters;
+import org.junit.Test;
+
+import java.util.Base64;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Simeon Macke
+ */
+public class Argon2EncodingUtilsTests {
+
+	private final Base64.Decoder decoder = Base64.getDecoder();
+
+	private TestDataEntry testDataEntry1 = new TestDataEntry(
+			"$argon2i$v=19$m=1024,t=3,p=2$Y1JkRmJDdzIzZ3oyTWx4aw$cGE5Cbd/cx7micVhXVBdH5qTr66JI1iUyuNNVAnErXs",
+			new Argon2EncodingUtils.Argon2Hash(decoder.decode("cGE5Cbd/cx7micVhXVBdH5qTr66JI1iUyuNNVAnErXs"),
+					(new Argon2Parameters.Builder(Argon2Parameters.ARGON2_i)).
+							withVersion(19).withMemoryAsKB(1024).withIterations(3).withParallelism(2).
+							withSalt("cRdFbCw23gz2Mlxk".getBytes()).build()
+			));
+
+	private TestDataEntry testDataEntry2 = new TestDataEntry(
+			"$argon2id$v=19$m=333,t=5,p=2$JDR8N3k1QWx0$+PrEoHOHsWkU9lnsxqnOFrWTVEuOh7ZRIUIbe2yUG8FgTYNCWJfHQI09JAAFKzr2JAvoejEpTMghUt0WsntQYA",
+			new Argon2EncodingUtils.Argon2Hash(decoder.decode("+PrEoHOHsWkU9lnsxqnOFrWTVEuOh7ZRIUIbe2yUG8FgTYNCWJfHQI09JAAFKzr2JAvoejEpTMghUt0WsntQYA"),
+					(new Argon2Parameters.Builder(Argon2Parameters.ARGON2_id)).
+							withVersion(19).withMemoryAsKB(333).withIterations(5).withParallelism(2).
+							withSalt("$4|7y5Alt".getBytes()).build()
+			));
+
+	@Test
+	public void decode_1_version() throws Exception {
```

Comment:
Please switch to using a test method naming scheme of `<method-under-test>When<condition>Then<Expectation>`

Reply:
is it alright now?